### ISSUE TITLE
No Response

### ISSUE THREAD

**Issue by muzzworld** [hannseman/homebridge-mi-hygrothermograph#17]
I feel like I've added this as I should and the device shows up in the Home app, and when I view details it lists the manufacturer/model number etc so I assume home bridge has picked it up (unless this is set by the plugin itself).  One thing I can't seem to do as I am on a mac is find out the address, but I shouldn't really need this as it is the only device of its type for my instance.
However the two devices (Temp + Humidity) both register "No Response".
I have hooked up the Mi Home app to the device and it shows both Temp and Humidity no problems.
I am using Apple Home and a mac mini 2012 running home bridge.

Any advice would be much appreciated!

**Comment by hannseman** [comment]
The listed details are actually set by the plugin itself so it's not certain that communication to the sensor have been established. 

Are you sure that the sensor is in range to the Mac Mini? The range of this sensor is actually pretty bad. Try placing it next to the Mac Mini and see if you get any values.

**Comment by muzzworld** [comment]
Thanks for your response and suggestion. I put the sensor directly next to the mac mini and still no response (although I didn't restart homebridge but not sure that would make a difference.)

If the range is so poor how do others integrate it into their homebridge setups?

**Comment by hannseman** [comment]
The range is not so poor that you need to keep it right next to the Homebridge device but it’s a good test to see if that’s the problem. I have some problems at my home with thicker walls. You shouldn’t have to restart homebridge to get any results when back in range so your problem is probably something else.

Could you turn on debug logging in homebridge with the ‘-D’-flag and paste the output here.

**Comment by muzzworld** [comment]
will do soon - appreciate your help!

**Comment by muzzworld** [comment]
{
    "bridge": {
        "name": "Homebridge",
        "username": "xxxx",
        "pin": "xxxx",
        "port": 51826
    },
    "accessories": [
        {
            "accessory": "Hygrotermograph",
            "name": "BOM",
            "timeout": 15
        },
        {
            "accessory": "ComputerSpeakers",
            "name": "maclounge",
            "services": [
                "speaker",
                "fan"
            ]
        }
    ],
    "platforms": [
        {
            "platform": "broadlinkPlatform",
            "accessories": [
                {
                    "name": "MP1",
                    "type": "MP",
                    "mac": "34:ea:34:d1:16:32",
                    "local_ip_address": "10.0.1.20"
                }
            ]
        },
        {
            "platform": "config",
            "name": "Config",
            "port": 8080,
            "sudo": true,
            "log": {
                "method": "file",
                "path": "/var/log/homebridge.log"
            }
        },
        {
            "platform": "XiaomiMiio",
            "name": "XiaomiMiio",
            "pollChanges": true,
            "pollInterval": 15,
            "searchInterval": 900
        },
        {
            "platform": "DACP",
            "devices": [
                {
                    "name": "iTunes",
                    "pairing": "7C69847445071C4B",
                    "serviceName": "4B8695C8552DE037",
                    "features": {
                        "alternate-playpause-switch": true,
                        "album-artwork": "/tmp/nowplaying.png"
                    }
                }
            ]
        }
    ]
}

**Comment by muzzworld** [comment]
[6/13/2018, 8:31:20 PM] Loaded config.json with 2 accessories and 4 platforms.
[6/13/2018, 8:31:20 PM] ---
[6/13/2018, 8:31:20 PM] Loaded plugin: homebridge-broadlink-platform
[6/13/2018, 8:31:20 PM] Registering platform 'homebridge-broadlink-platform.broadlinkPlatform'
[6/13/2018, 8:31:20 PM] ---
[6/13/2018, 8:31:20 PM] Loaded plugin: homebridge-config-ui-x
[6/13/2018, 8:31:20 PM] Registering platform 'homebridge-config-ui-x.config'
[6/13/2018, 8:31:20 PM] ---
[6/13/2018, 8:31:20 PM] Loaded plugin: homebridge-dacp
[6/13/2018, 8:31:20 PM] Registering platform 'homebridge-dacp.DACP'
[6/13/2018, 8:31:20 PM] ---
[6/13/2018, 8:31:20 PM] Loaded plugin: homebridge-mi-hygrothermograph
[6/13/2018, 8:31:21 PM] Registering accessory 'homebridge-mi-hygrothermograph.Hygrotermograph'
[6/13/2018, 8:31:21 PM] ---
[6/13/2018, 8:31:21 PM] Loaded plugin: homebridge-mi-philips-light
[6/13/2018, 8:31:21 PM] ---
[6/13/2018, 8:31:21 PM] Loaded plugin: homebridge-miio
[6/13/2018, 8:31:21 PM] Registering platform 'homebridge-miio.XiaomiMiio'
[6/13/2018, 8:31:21 PM] ---
[6/13/2018, 8:31:21 PM] Loaded plugin: homebridge-pc-volume
[6/13/2018, 8:31:21 PM] Registering accessory 'homebridge-pc-volume.ComputerSpeakers'
[6/13/2018, 8:31:21 PM] ---
[6/13/2018, 8:31:21 PM] Loading 4 platforms...
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Initializing broadlinkPlatform platform...
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Created MP1s1 s1 Accessory
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Created MP1s2 s2 Accessory
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Created MP1s3 s3 Accessory
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Created MP1s4 s4 Accessory
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Initializing platform accessory 'MP1s1'...
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Initializing platform accessory 'MP1s2'...
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Initializing platform accessory 'MP1s3'...
[6/13/2018, 8:31:21 PM] [broadlinkPlatform] Initializing platform accessory 'MP1s4'...
[6/13/2018, 8:31:21 PM] [Config] Initializing config platform...
[6/13/2018, 8:31:21 PM] [Config] Spawning homebridge-config-ui-x with PID 5675
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Initializing XiaomiMiio platform...
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Setting up Miio platform
[6/13/2018, 8:31:21 PM] [DACP] Initializing DACP platform...
[6/13/2018, 8:31:21 PM] [DACP] DACP Platform Plugin Loaded - Version 0.9.2
[6/13/2018, 8:31:21 PM] [DACP] Found accessory in config: "iTunes"
[6/13/2018, 8:31:21 PM] [DACP] Initializing platform accessory 'iTunes'...
[6/13/2018, 8:31:21 PM] Loading 2 accessories...
[6/13/2018, 8:31:21 PM] [BOM] Initializing Hygrotermograph accessory...
[6/13/2018, 8:31:21 PM] [BOM] Initialized accessory
[6/13/2018, 8:31:21 PM] [maclounge] Initializing ComputerSpeakers accessory...
[6/13/2018, 8:31:21 PM] [XiaomiMiio] miIO Plug 66071145 Configure Accessory
[6/13/2018, 8:31:21 PM] [XiaomiMiio] miIO Plug 61427310 Configure Accessory
[6/13/2018, 8:31:21 PM] [XiaomiMiio] miIO Plug 66090887 Configure Accessory
[6/13/2018, 8:31:21 PM] [XiaomiMiio] miIO Plug 66089503 Configure Accessory
[6/13/2018, 8:31:21 PM] [XiaomiMiio] miIO Plug 61387147 Configure Accessory
Setup Payload:
X-HM://
Scan this code with your HomeKit app on your iOS device to pair with Homebridge:

Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ xxxxxxxxx│     
    └────────────┘     
                       
[6/13/2018, 8:31:21 PM] [XiaomiMiio] DidFinishLaunching
[6/13/2018, 8:31:21 PM] [DACP] Starting DACP browser...
[6/13/2018, 8:31:21 PM] Homebridge is running on port 51826.
[6/13/2018, 8:31:21 PM] iTunes Artwork is running on port 51456.
[6/13/2018, 8:31:21 PM] [XiaomiMiio] device discovered 61387147
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Investigating Miio Device at udp://10.0.1.85:54321
[6/13/2018, 8:31:21 PM] [XiaomiMiio] device discovered 61826023
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Investigating Miio Device at udp://10.0.1.213:54321
[6/13/2018, 8:31:21 PM] [XiaomiMiio] device discovered 61778439
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Investigating Miio Device at udp://10.0.1.212:54321
[6/13/2018, 8:31:21 PM] [XiaomiMiio] device discovered 66071145
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Investigating Miio Device at udp://10.0.1.79:54321
[6/13/2018, 8:31:21 PM] [XiaomiMiio] device discovered 61427310
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Investigating Miio Device at udp://10.0.1.83:54321
[6/13/2018, 8:31:21 PM] [XiaomiMiio] device discovered 66090887
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Investigating Miio Device at udp://10.0.1.84:54321
[6/13/2018, 8:31:21 PM] [XiaomiMiio] device discovered 66089503
[6/13/2018, 8:31:21 PM] [XiaomiMiio] Investigating Miio Device at udp://10.0.1.80:54321
[6/13/2018, 8:31:22 PM] [BOM] Stopped scanning
[6/13/2018, 8:31:23 PM] [DACP] The accessory iTunes is announced as 10.0.1.105:3689.
[6/13/2018, 8:31:23 PM] [DACP] It's an iTunes named muzminitunes
[6/13/2018, 8:31:23 PM] [DACP] Attempting to reconnect to iTunes in 0.1 seconds.
[0;37m[6/13/2018, 8:31:23 PM][0m [0;36m[Config][0m Using Form Authentication
[6/13/2018, 8:31:23 PM] [DACP] Connecting to iTunes (10.0.1.105:3689)
[6/13/2018, 8:31:23 PM] [DACP] Creating status connection to 10.0.1.105:3689
[0;37m[6/13/2018, 8:31:23 PM][0m [0;36m[Config][0m Console v3.8.3 is listening on port 8080.
Established connection to 10.0.1.105:3689 with session ID 808541711
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'aeIM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
[6/13/2018, 8:31:23 PM] [DACP] Connected to muzminitunes
Skipping 'ceNR' - don't know how to parse. length=11
Skipping 'aeaw' - don't know how to parse. length=1
[6/13/2018, 8:31:23 PM] [DACP] Creating properties connection to 10.0.1.105:3689
[6/13/2018, 8:31:23 PM] [DACP] Updating characteristics with current volume: v=86
[6/13/2018, 8:31:28 PM] [XiaomiMiio] Couldn't investigate device { id: '61826023',
  address: '10.0.1.213',
  port: 54321,
  token: '3f2ecaf6902b65cc899d805af4b73e5b',
  autoToken: false,
  lastSeen: 1528885881736,
  writeOnly: true } { Error: Call to device timed out
    at Timeout.retry [as _onTimeout] (/usr/local/lib/node_modules/homebridge-miio/node_modules/miio/lib/device.js:225:18)
    at ontimeout (timers.js:458:11)
    at tryOnTimeout (timers.js:296:5)
    at Timer.listOnTimeout (timers.js:259:5) code: 'timeout' }
[6/13/2018, 8:31:28 PM] [XiaomiMiio] Couldn't investigate device { id: '61778439',
  address: '10.0.1.212',
  port: 54321,
  token: 'e87116e6acc53f230a5a52ec503b671f',
  autoToken: false,
  lastSeen: 1528885881736,
  writeOnly: true } { Error: Call to device timed out
    at Timeout.retry [as _onTimeout] (/usr/local/lib/node_modules/homebridge-miio/node_modules/miio/lib/device.js:225:18)
    at ontimeout (timers.js:458:11)
    at tryOnTimeout (timers.js:296:5)
    at Timer.listOnTimeout (timers.js:259:5) code: 'timeout' }
(node:5671) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'model' of undefined
    at miio.device.catch.then (/usr/local/lib/node_modules/homebridge-miio/index.js:176:48)
    at <anonymous>
(node:5671) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:5671) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:5671) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'model' of undefined
    at miio.device.catch.then (/usr/local/lib/node_modules/homebridge-miio/index.js:176:48)
    at <anonymous>
(node:5671) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
[6/13/2018, 8:31:37 PM] Got SIGTERM, shutting down Homebridge...

**Comment by hannseman** [comment]
Thanks! Found something interesting in the logs:

[6/13/2018, 8:31:22 PM] [BOM] Stopped scanning

I don’t know why it stops scanning for Bluetooth devices. I’ll have to extend the logging to get more information about that. I’ll do some more reading on why noble might behave like this.

I’ve successfully tried running this plugin on a MacBook Pro 15” Mid 2012 with High Sierra. What MacOS are you running?

**Comment by muzzworld** [comment]
good pickup! I missed that.

I am using the latest OS on a mid 2012 macmini.

**Comment by muzzworld** [comment]
Hi hannseman, any more ideas on this issue? I want to purchase more of these devices but won't be doing that if they don't work with homebridge!
I am also looking at buying the BLE gateway (the Xiaomi Bedside Lamp) - will that affect this plugin?

Cheers!

**Comment by hannseman** [comment]
Hi! I just released a new version with a bit more detailed logging. If you update and could try and run the logging again and find the entry where it says `Stop scanning` and paste that here maybe we could get a hint about what might be causing your problems.

I'm running this plugin on a Raspberry Pi with no problems at all so that might be a workaround if you are willing to do that. But as I've said this plugin *should* work on MacOS-devices.

Xiaomi Bedside Lamp shouldn't cause any trouble for this plugin but I've never tried one.

**Comment by muzzworld** [comment]
Thanks hannes!

I updated and restarted. Here are the logs now.

**Comment by muzzworld** [comment]
[6/20/2018, 7:37:20 AM] [broadlinkPlatform] checking status for MP1s4
[6/20/2018, 7:37:20 AM] [XiaomiMiio] miIO Plug 66071145 fetch status
[6/20/2018, 7:37:20 AM] [broadlinkPlatform] checking status for MP1s1
[6/20/2018, 7:37:21 AM] [broadlinkPlatform] MP1s2 power is on - false
[6/20/2018, 7:37:21 AM] [broadlinkPlatform] MP1s3 power is on - false
[6/20/2018, 7:37:21 AM] [broadlinkPlatform] MP1s4 power is on - false
[6/20/2018, 7:37:22 AM] [broadlinkPlatform] MP1s1 power is on - false
[6/20/2018, 7:51:00 AM] [DACP] Returning current playback state: paused
[6/20/2018, 7:51:00 AM] [XiaomiMiio] miIO Plug 66089503 fetch status
[6/20/2018, 7:51:00 AM] [XiaomiMiio] miIO Plug 61387147 fetch status
[6/20/2018, 7:51:00 AM] [broadlinkPlatform] checking status for MP1s2
[6/20/2018, 7:51:00 AM] [XiaomiMiio] miIO Plug 66090887 fetch status
[6/20/2018, 7:51:00 AM] [broadlinkPlatform] checking status for MP1s3
[6/20/2018, 7:51:00 AM] [XiaomiMiio] miIO Plug 61427310 fetch status
[6/20/2018, 7:51:00 AM] [broadlinkPlatform] checking status for MP1s4
[6/20/2018, 7:51:00 AM] [XiaomiMiio] miIO Plug 66071145 fetch status
[6/20/2018, 7:51:00 AM] [broadlinkPlatform] checking status for MP1s1
[6/20/2018, 7:51:01 AM] [broadlinkPlatform] MP1s2 power is on - false
[6/20/2018, 7:51:01 AM] [broadlinkPlatform] MP1s3 power is on - false
[6/20/2018, 7:51:01 AM] [broadlinkPlatform] MP1s4 power is on - false
[6/20/2018, 7:51:01 AM] [broadlinkPlatform] MP1s1 power is on - false
[6/20/2018, 7:51:06 AM] [DACP] Returning current playback state: paused
[6/20/2018, 7:51:06 AM] [XiaomiMiio] miIO Plug 66089503 fetch status
[6/20/2018, 7:51:06 AM] [XiaomiMiio] miIO Plug 61387147 fetch status
[6/20/2018, 7:51:06 AM] [broadlinkPlatform] checking status for MP1s2
[6/20/2018, 7:51:06 AM] [XiaomiMiio] miIO Plug 66090887 fetch status
[6/20/2018, 7:51:06 AM] [broadlinkPlatform] checking status for MP1s3
[6/20/2018, 7:51:06 AM] [XiaomiMiio] miIO Plug 61427310 fetch status
[6/20/2018, 7:51:06 AM] [broadlinkPlatform] checking status for MP1s4
[6/20/2018, 7:51:06 AM] [XiaomiMiio] miIO Plug 66071145 fetch status
[6/20/2018, 7:51:06 AM] [broadlinkPlatform] checking status for MP1s1
[6/20/2018, 7:51:07 AM] [broadlinkPlatform] MP1s2 power is on - false
[6/20/2018, 7:51:07 AM] [broadlinkPlatform] MP1s4 power is on - false
[6/20/2018, 7:51:09 AM] [broadlinkPlatform] MP1s1 power is on - false
[6/20/2018, 7:51:10 AM] [broadlinkPlatform] MP1s3 power is on - false
[6/20/2018, 7:54:40 AM] Got SIGTERM, shutting down Homebridge...
[6/20/2018, 7:54:42 AM] Loaded config.json with 2 accessories and 4 platforms.
[6/20/2018, 7:54:42 AM] ---
[6/20/2018, 7:54:42 AM] Loaded plugin: homebridge-broadlink-platform
[6/20/2018, 7:54:42 AM] Registering platform 'homebridge-broadlink-platform.broadlinkPlatform'
[6/20/2018, 7:54:42 AM] ---
[6/20/2018, 7:54:42 AM] Loaded plugin: homebridge-config-ui-x
[6/20/2018, 7:54:42 AM] Registering platform 'homebridge-config-ui-x.config'
[6/20/2018, 7:54:42 AM] ---
[6/20/2018, 7:54:42 AM] Loaded plugin: homebridge-dacp
[6/20/2018, 7:54:42 AM] Registering platform 'homebridge-dacp.DACP'
[6/20/2018, 7:54:42 AM] ---
[6/20/2018, 7:54:42 AM] Loaded plugin: homebridge-mi-hygrothermograph
[6/20/2018, 7:54:43 AM] Registering accessory 'homebridge-mi-hygrothermograph.Hygrotermograph'
[6/20/2018, 7:54:43 AM] ---
[6/20/2018, 7:54:43 AM] Loaded plugin: homebridge-mi-philips-light
[6/20/2018, 7:54:43 AM] ---
[6/20/2018, 7:54:43 AM] Loaded plugin: homebridge-miio
[6/20/2018, 7:54:43 AM] Registering platform 'homebridge-miio.XiaomiMiio'
[6/20/2018, 7:54:43 AM] ---
[6/20/2018, 7:54:43 AM] Loaded plugin: homebridge-pc-volume
[6/20/2018, 7:54:43 AM] Registering accessory 'homebridge-pc-volume.ComputerSpeakers'
[6/20/2018, 7:54:43 AM] ---
[6/20/2018, 7:54:43 AM] Loading 4 platforms...
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Initializing broadlinkPlatform platform...
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Created MP1s1 s1 Accessory
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Created MP1s2 s2 Accessory
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Created MP1s3 s3 Accessory
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Created MP1s4 s4 Accessory
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Initializing platform accessory 'MP1s1'...
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Initializing platform accessory 'MP1s2'...
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Initializing platform accessory 'MP1s3'...
[6/20/2018, 7:54:43 AM] [broadlinkPlatform] Initializing platform accessory 'MP1s4'...
[6/20/2018, 7:54:43 AM] [Config] Initializing config platform...
[6/20/2018, 7:54:43 AM] [Config] Spawning homebridge-config-ui-x with PID 24667
[6/20/2018, 7:54:43 AM] [XiaomiMiio] Initializing XiaomiMiio platform...
[6/20/2018, 7:54:43 AM] [XiaomiMiio] Setting up Miio platform
[6/20/2018, 7:54:43 AM] [DACP] Initializing DACP platform...
[6/20/2018, 7:54:43 AM] [DACP] DACP Platform Plugin Loaded - Version 0.9.2
[6/20/2018, 7:54:43 AM] [DACP] Found accessory in config: "iTunes"
[6/20/2018, 7:54:43 AM] [DACP] Initializing platform accessory 'iTunes'...
[6/20/2018, 7:54:43 AM] Loading 2 accessories...
[6/20/2018, 7:54:43 AM] [BOM] Initializing Hygrotermograph accessory...
[6/20/2018, 7:54:43 AM] [maclounge] Initializing ComputerSpeakers accessory...
[6/20/2018, 7:54:43 AM] [XiaomiMiio] miIO Plug 66071145 Configure Accessory
[6/20/2018, 7:54:43 AM] [XiaomiMiio] miIO Plug 61427310 Configure Accessory
[6/20/2018, 7:54:43 AM] [XiaomiMiio] miIO Plug 66090887 Configure Accessory
[6/20/2018, 7:54:43 AM] [XiaomiMiio] miIO Plug 66089503 Configure Accessory
[6/20/2018, 7:54:43 AM] [XiaomiMiio] miIO Plug 61387147 Configure Accessory
Setup Payload:
X-HM://0023ISYWYMOE2
Scan this code with your HomeKit app on your iOS device to pair with Homebridge:


Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:

    ┌────────────┐
    │ xxxxxxxxx │
    └────────────┘

[6/20/2018, 7:54:43 AM] [XiaomiMiio] DidFinishLaunching
[6/20/2018, 7:54:43 AM] [DACP] Starting DACP browser...
[6/20/2018, 7:54:43 AM] Homebridge is running on port 51826.
[6/20/2018, 7:54:43 AM] iTunes Artwork is running on port 54517.
[6/20/2018, 7:54:43 AM] [BOM] Stop scanning. (unsupported)
[6/20/2018, 7:54:44 AM] [Config] Using Form Authentication
[6/20/2018, 7:54:44 AM] [Config] Console v3.8.5 is listening on port 8080.
[6/20/2018, 7:54:44 AM] [DACP] The accessory iTunes is announced as 10.0.1.105:3689.
[6/20/2018, 7:54:44 AM] [DACP] It's an iTunes named muzminitunes
[6/20/2018, 7:54:44 AM] [DACP] Attempting to reconnect to iTunes in 0.1 seconds.
[6/20/2018, 7:54:44 AM] [DACP] Connecting to iTunes (10.0.1.105:3689)
[6/20/2018, 7:54:44 AM] [DACP] Creating status connection to 10.0.1.105:3689
Established connection to 10.0.1.105:3689 with session ID 130867696
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'aeIM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
Skipping 'mdbk' - don't know how to parse. length=4
Skipping 'aeHS' - don't know how to parse. length=1
Skipping 'aeCs' - don't know how to parse. length=4
Skipping 'mrpr' - don't know how to parse. length=8
Skipping 'aeRM' - don't know how to parse. length=8
Skipping 'meds' - don't know how to parse. length=4
[6/20/2018, 7:54:45 AM] [DACP] Connected to muzminitunes
Skipping 'ceNR' - don't know how to parse. length=11
Skipping 'aeaw' - don't know how to parse. length=1
[6/20/2018, 7:54:45 AM] [DACP] Creating properties connection to 10.0.1.105:3689
[6/20/2018, 7:54:45 AM] [DACP] Updating characteristics with current volume: v=86

**Comment by hannseman** [comment]
Thanks!

`[6/20/2018, 7:54:43 AM] [BOM] Stop scanning. (unsupported)`

AFAIK `unsupported` is returned when the system does not support Bluetooth Low Energy:
https://github.com/noble/noble/issues/568
https://github.com/noble/noble/issues/105

Could you check System Report -> Hardware -> Bluetooth and look for Bluetooth Low Energy Supported: No.

If this is the case then your only option is to buy a Bluetooth 4.0 adapter or system supporting Bluetooth Low Energy.

**Comment by muzzworld** [comment]
You are a genius - I have 2 mac minis but one is a bit earlier than the other, needless to say it is the one that doesn't support BLE and is also the one I chose to put homebridge on.

I will now move my homebridge server to the newer one and reckon that will work. I thank you for your work in developing this plug in and also helping me fault find - sorry if I have wasted your time!

I will let you know how it goes on the newer server once I have it up and running.

Cheers!
